web/src/main/java/org/cbioportal/genome_nexus/web/validation/GenomicLocationValidator.java

```diff
@@ -0,0 +1,29 @@
+package org.cbioportal.genome_nexus.web.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.cbioportal.genome_nexus.web.validation.ValidGenomicLocation;
+
+public class GenomicLocationValidator implements ConstraintValidator<ValidGenomicLocation, String> {
+    
+    public void initialize(ValidGenomicLocation constraintAnnotation) {
+    }
+
+    public boolean isValid(String genomicLocation, ConstraintValidatorContext context) 
+    {
+        String[] genomicLocationSplit = genomicLocation.split(",");
+        if (genomicLocationSplit.length <5){
+            return false;
+        }
+        else{
+            boolean result = true;
+            // chr [1,24] or X or Y
+            result &= genomicLocationSplit[0].matches("X") || genomicLocationSplit[0].matches("Y") || genomicLocationSplit[0].matches("[0-9]+") && Integer.valueOf(genomicLocationSplit[0]) >= 1 && Integer.valueOf(genomicLocationSplit[0]) <= 24;
```

Comment:
you should break long lines of conditions into multiple lines for better readability, for example something like this:

```suggestion
            result &= genomicLocationSplit[0].matches("X") || genomicLocationSplit[0].matches("Y") ||
             (genomicLocationSplit[0].matches("[0-9]+") && Integer.valueOf(genomicLocationSplit[0]) >= 1 && Integer.valueOf(genomicLocationSplit[0]) <= 24);
```